crates/cheatnet/src/lib.rs

```diff
@@ -1,30 +1,46 @@
-use blockifier::state::cached_state::CachedState;
+use crate::forking::state::ForkStateReader;
+use crate::forking::worker::Worker;
+use crate::state::CustomStateReader;
+use blockifier::state::cached_state::{CachedState, GlobalContractCache};
 use camino::Utf8PathBuf;
 use constants::build_testing_state;
+use starknet::core::types::BlockId;
 use starknet_api::hash::StarkFelt;
 use starknet_api::transaction::ContractAddressSalt;
-use state::{CheatcodeState, DictStateReader};
+use state::CheatcodeState;
 
 pub mod cheatcodes;
 pub mod constants;
 pub mod contract_print;
 pub mod conversions;
+pub mod forking;
 pub mod panic_data;
 pub mod rpc;
 pub mod state;
 
 pub struct CheatnetState {
     cheatcode_state: CheatcodeState,
-    blockifier_state: CachedState<DictStateReader>,
+    blockifier_state: CachedState<CustomStateReader>,
     pub deploy_salt_base: u32,
 }
 
 impl CheatnetState {
     #[must_use]
-    pub fn new(predeployed_contracts: &Utf8PathBuf) -> Self {
+    pub fn new(predeployed_contracts: &Utf8PathBuf, rpc_config: Option<(&str, BlockId)>) -> Self {
+        let dict_state_reader = build_testing_state(predeployed_contracts);
         CheatnetState {
             cheatcode_state: CheatcodeState::new(),
-            blockifier_state: build_testing_state(predeployed_contracts),
+            blockifier_state: CachedState::new(
```

Comment:
I still think that `StateReader` should be provided from outside rather than passing optional RPC conf

Reply:
If you dislike that approach i suggest as well an alternative (which i kind of dislike as well, but not as much as this one):

Add another constructor method which instantiates the whole thing, but name it `from_rpc` or `new_forked` and then we can differentiate on how we instantiate those.

That said - I still prefer injecting such deps since it's cleaner and this code will actually compose more nicely with changes that will inevitably be made to it (multifork)